### PR TITLE
Users with monitor privileges can access async_search/status endpoint even when setting keep_alive

### PR DESCRIPTION
**Pull request by quux00**
Fixes a bug in the async-search status endpoint where a user with monitor privileges
is not able to access the status endpoint when setting keep_alive state of the async-search.

### CODE REVIEW

**Comment by quux00** [comment]
The test that found this bug is documented here: https://github.com/elastic/elasticsearch/pull/106638#issuecomment-2015555180

This PR is based on [a patch](https://github.com/elastic/elasticsearch/pull/106638#pullrequestreview-1964122698) recommended by @albertzaharovits

**Comment by quux00** [comment]
**UPDATE**: Upon testing I see that this change doesn't fix the bug - it actually makes it worse. Now a user with only monitor privileges can modify the keep_alive status both while it is running (what this change allows) and after it has completed running. Whereas in main, it is only allowed after the search has completed. So I recommend closing this PR and not merging.

**Comment by albertzaharovits** [comment]
> Upon testing I see that this change doesn't fix the bug - it actually makes it worse. Now a user with only monitor privileges can modify the keep_alive status both while it is running (what this change allows) and after it has completed running. Whereas in main, it is only allowed after the search has completed. So I recommend closing this PR and not merging.

@quux00 What would you propose the behavior for the user with only the `monitor` cluster privilege should be, when using the `keep_alive` parameter, when retrieving the status of non-owned async searches?
Putting myself in the user's shoes, I firmly believe that the behavior depending on whether the async search is still running or has completed is fickle, which limits its usefulness in practice. As a dev, I don't think there is case to be made that modifying the keep alive of in-progress or completed async searches requires different permissions, especially given the current coarseness of ES's permission model.

**Comment by quux00** [comment]
> What would you propose the behavior for the user with only the `monitor` cluster privilege should be, when using the `keep_alive` parameter, when retrieving the status of non-owned async searches? Putting myself in the user's shoes, I firmly believe that the behavior depending on whether the async search is still running or has completed is fickle, which limits its usefulness in practice. As a dev, I don't think there is case to be made that modifying the keep alive of in-progress or completed async searches requires different permissions, especially given the current coarseness of ES's permission model.

My thinking was that users with only `monitor` status (that did not create the async-search request) shouldn't be allowed to modify the search in any way. The [`keep_alive` flag](https://www.elastic.co/guide/en/elasticsearch/reference/current/async-search.html#get-async-search-status) modifies the how long a completed search is retained in the .async-search system index. It could be disruptive to a user for an admin to be able to change how long a search result is kept. However, you could argue that this is an admin's prerogative, maybe to save disk space. Kibana routinely sets the keep_alive flag to a much lower setting than the default to minimize excess disk usage, so perhaps we should allow it then and go ahead and merge this ticket.

File: x-pack/plugin/async-search/qa/security/src/javaRestTest/java/org/elasticsearch/xpack/search/AsyncSearchSecurityIT.java [hunk 1]
```diff
@@ -250,6 +249,9 @@ private static void userBasedPermissionsAsserts(String user, String other, Strin
         // user-monitor can access the status
         assertOK(getAsyncStatus(id, "user-monitor"));
 
+        // user-monitor can access status and set keep_alive
+        assertOK(getAsyncStatusAndSetKeepAlive(id, "user-monitor"));
```
Comment (quux00) [inline x-pack/plugin/async-search/qa/security/src/javaRestTest/java/org/elasticsearch/xpack/search/AsyncSearchSecurityIT.java:253]: Without the corresponding change in TransportGetAsyncStatusAction.java, this call (when called from `testStatusWithUsersWhileSearchIsRunning`) fails with:

```
org.elasticsearch.client.ResponseException: method [GET], host [http://[::1]:55416], URI [/_async_search/status/Fm8wcUlRZzJvU21HaWhNa0lfOEJHVmcabTRBaE4zemxTdnl0TGotODlnTWpyZzo1NjI=?keep_alive=3m], status line [HTTP/1.1 404 Not Found]
{"error":{"root_cause":[{"type":"resource_not_found_exception","reason":"Fm8wcUlRZzJvU21HaWhNa0lfOEJHVmcabTRBaE4zemxTdnl0TGotODlnTWpyZzo1NjI="}],"type":"resource_not_found_exception","reason":"Fm8wcUlRZzJvU21HaWhNa0lfOEJHVmcabTRBaE4zemxTdnl0TGotODlnTWpyZzo1NjI="},"status":404}
```

**Review by albertzaharovits — APPROVED** [review]
LGTM

**Pull request merged**